### PR TITLE
Add tests and update csg boltzmann

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
In an attempt to remove use of histogram and replace it with histogramnew without breaking the code it was necessary to setup a few unit tests for classes that made use of the histogram class. This merge is responsible for creating these tests which are located in the csg-boltzmann folder. 
In response to issue https://github.com/votca/tools/issues/47

### CODE REVIEW

**Comment by junghans** [comment]
Does that fix #155?

**Comment by junghans** [comment]
And could we add a regression test for csg_boltzmann? I.e echoing some commands into the command and compare the output.

**Comment by JoshuaSBrown** [comment]
The boltzmann constant used in the boltzmann inversion function in the tabulated potential has a hard coded value in what I believe is units of [ kJ/(mol K) ]. Is this what you are referring too?

**Comment by JoshuaSBrown** [comment]
It is not really that clear to me how to use csg_boltzmann. @junghans in the interactive mode I can specify a histogram etc... but is there a way to do this without going into the interactive mode? I'm also not sure what to use for the selection to get meaningful output? I've been trying with files in tools/references/spce but I don't know if this is appropriate either. 

    list: list all available bonds
    vals <file> <selection>: write values to file
    hist <file> <selection>: create histogram
    tab <file> <selection>: create tabulated potential
    autocor <file> <selection>: calculate autocorrelation, only one row allowed in selection!
    cor <file> <selection>: calculate correlations, first row is correlated with all other rows

**Comment by junghans** [comment]
I was thinking about something like `echo "command1\ncommand2\nq" | csg_boltzmann`.

**Comment by junghans** [comment]
I have did something like:
```
hist end.txt end-end
vals end-v.txt end-end
hist AA.txt AA-bond
hist AAA.txt AAA-angle
hist set periodic 1
hist set auto 0
hist set min -3.14159
hist set max 3.14159
hist AAAA.txt AAAA-dihedral
tab set T 400
tab U_AA.txt AA-bond
tab U_AAA.txt AAA-angle
tab set periodic 1
tab set smooth1 1
tab set auto 0
tab set min -3.14159
tab set max 3.14159
tab U_AAAA.txt AAAA-dihedral
autocor end-cor.txt end-end
autocor AA-cor.txt AA-bond[4]
autocor AAA-cor.txt AAA-angle[5]
autocor AAAA-cor.txt AAAA-dihedral[4]
cor bond-angle.txt AAA-angle[4] AA-bond
cor bond-dihe.txt AAAA-dihedral[4] AA-bond
cor angle-dihe.txt AAAA-dihedral[4] AAA-angle
q
```
and then `cat "cmd" | csg_boltzmann --top topol.tpr` before.

However it seems CMake has a hard time to redirect stdin into tests (see https://stackoverflow.com/questions/36304289/how-to-use-redirection-in-cmake-add-test).

So @JoshuaSBrown how hard would it be to add an option (e.g. `--in` to have `csg_boltzmann` read from a file instead of stdin?

**Comment by JoshuaSBrown** [comment]
I can do that but let's put it in a separate merge, that will be more than a quick fix, this merge is good to go unless you see something else that I have done that should be changed?

**Comment by junghans** [comment]
If you do that than I will write the tests!

File: src/csg_boltzmann/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 file(GLOB CSG_BO_SOURCES *.cc)
+add_library(votca_csg_boltzmann ${CSG_BO_SOURCES})
```
Comment (junghans) [inline src/csg_boltzmann/CMakeLists.txt]: Won't that library have a `main()` if we use all `csg_boltzmann` sources?

**Comment by junghans** [comment]
Will this help with https://github.com/votca/tools/issues/47?

**Comment by junghans** [comment]
Maybe we can add an option to change the unit of `kB` down the road?

File: src/csg_boltzmann/CMakeLists.txt (hunk 1 above)
Comment (JoshuaSBrown) [reply src/csg_boltzmann/CMakeLists.txt]: Yes, I guess I could remove the main.cc that would be less sloppy.

**Comment by JoshuaSBrown** [comment]
Yes, that could be a separate issue. How units are handled could be streamlined across the whole VOTCA package. Though I don't yet have a good feel for that aspect of the VOTCA package yet as I have not really used the tools with the various file formats we support.

File: src/csg_boltzmann/tabulatedpotential.cc [hunk 2]
```diff
@@ -15,333 +15,348 @@
...
-#include "tabulatedpotential.h"
+#include <votca/tools/histogram.h>
 
 using namespace std;
 using namespace boost;
+using namespace votca::tools;
+using namespace votca::csg;
+
+/* Boltzmann constant in units [ kJ/(mol K) ] */
+const double kB = 0.0083109;
```
Comment (JensWehner) [inline src/csg_boltzmann/tabulatedpotential.cc]: move to votca/tools/constants

File: src/csg_boltzmann/tabulatedpotential.cc [hunk 3]
```diff
@@ -15,333 +15,348 @@
...
+
+  _min = numeric_limits<double>::max();
+  _max = numeric_limits<double>::min();
+
+  for (size_t i = 0; i < data.size(); i++) {
+    _max = max(data[i], _max);
+    if (data[i] > 0)
+      _min = min(data[i], _min);
+  }
+  _max = -kB * _Temperature * log(_max);
```
Comment (JensWehner) [inline src/csg_boltzmann/tabulatedpotential.cc]: really good catch, well done

**Review by JensWehner — APPROVED** [review]
lgtm nice rework.  So we can remove histogram and rename histogram_new to histogram?

File: src/csg_boltzmann/tabulatedpotential.cc (hunk 3 above)
Comment (JoshuaSBrown) [reply src/csg_boltzmann/tabulatedpotential.cc]: Well you know I try... :)

File: src/csg_boltzmann/tabulatedpotential.cc (hunk 2 above)
Comment (JoshuaSBrown) [reply src/csg_boltzmann/tabulatedpotential.cc]: Sounds good.

**Comment by JoshuaSBrown** [comment]
@JensWehner and yes that is the idea, once we have good tests in place we should be able to replace histogram with histogramnew and then remove it hopefully without any hiccups.

**Comment by JoshuaSBrown** [comment]
https://github.com/votca/tools/pull/84 should be merged first.

**Pull request merged**